src/group_theory/free_abelian_group_finsupp.lean

```diff
@@ -0,0 +1,145 @@
+/-
+Copyright (c) 2021 Johan Commelin. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Johan Commelin
+-/
+
+import group_theory.free_abelian_group
+import data.finsupp.basic
+
+/-!
+# Isomorphism between `free_abelian_group X` and `X →₀ ℤ`
+
+In this file we construct the canonical isomorphism between `free_abelian_group X` and `X →₀ ℤ`.
+We use this to transport the notion of `support` from `finsupp` to `free_abelian_group`.
+
+## Main declarations
+
+- `free_abelian_group.equiv_finsupp`: group isomorphism between `free_abelian_group X` and `X →₀ ℤ`
+- `free_abelian_group.coeff`: the multiplicity of `x : X` in `a : free_abelian_group X`
+- `free_abelian_group.support`: the finset of `x : X` that occur in `a : free_abelian_group X`
```

Comment:
```suggestion
- `free_abelian_group.support`: the finset of `x : X` that occur in `a : free_abelian_group X`
+ `free_abelian_group.support`: the finset of `x : X` that occurs in `a : free_abelian_group X`
```

Reply:
I think `occur` is correct in this case... English is weird.

Reply:
Oh sure: the finset of all `x`'s... Shame on me!

Reply:
Well, I actually thought about this a bit, and have a small remark. I don't think there is anything weird with English here, it is a very simple and plain 3rd person singular which takes an `s`. The reason I got it wrong is that the sentence might be a bit short: I would say
- let A be the subset of R that contains/that intersects/that consists/that minimizes/...
- let A be the set of all `a : R` that are positive/that annihilate f/that minimize the gradient...
I think that "the finset of all `x : X` that occur in ..." would be nicer (a part from not having trapped me, but this is an aside ;), or  (longer) "the finset consisting of all `x: X` that occur in...".